django/contrib/postgres/constraints.py

```diff
@@ -206,9 +206,16 @@ def validate(self, model, instance, exclude=None, using=DEFAULT_DB_ALIAS):
                     self.get_violation_error_message(), code=self.violation_error_code
                 )
         else:
-            if (self.condition & Exists(queryset.filter(self.condition))).check(
-                replacement_map, using=using
-            ):
-                raise ValidationError(
-                    self.get_violation_error_message(), code=self.violation_error_code
-                )
+            against = instance._get_field_expression_map(
+                meta=model._meta, exclude=exclude
+            )
+            try:
+                if (self.condition & Exists(queryset.filter(self.condition))).check(
+                    against, using=using
+                ):
+                    raise ValidationError(
+                        self.get_violation_error_message(),
+                        code=self.violation_error_code,
+                    )
+            except FieldError:
+                pass
```

Comment:
IMO, we should use existing hook to check `self.condition` instead of catching `FieldError`, e.g.
```suggestion
            if exclude and self._expression_refs_exclude(model, self.condition, exclude):
                return
            if (self.condition & Exists(queryset.filter(self.condition))).check(
                replacement_map, using=using
            ):  
                raise ValidationError(
                    self.get_violation_error_message(), code=self.violation_error_code
                )
```